chips/sam4l/src/dma.rs

```diff
@@ -144,12 +153,18 @@ impl DMAChannel {
         DMAChannel {
             registers: (DMA_BASE_ADDR + (channel as usize) * DMA_CHANNEL_SIZE) as *mut DMARegisters,
             nvic: nvic,
-            client: None,
+            client: Cell::new(None),
+            width: Cell::new(DMAWidth::Width8Bit),
             enabled: Cell::new(false),
             buffer: TakeCell::empty(),
         }
     }
 
+    pub fn initialize(&self, client: &'static mut DMAClient, width: DMAWidth) {
```

Comment:
Should that assumption be guarded with a `if self.client is not None => panic!` here?